crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -436,6 +436,12 @@ impl CheckpointProposalContents {
     }
 }
 
+/// This is a message validators publish to consensus in order to sign checkpoint
+#[derive(Clone, Debug, Serialize, Deserialize, Hash)]
+pub struct SignedCheckpointData {
```

Comment:
I don't quite understand why you need a wrapper of `SignedCheckpointSummary`?

Reply:
Mostly because I am planning to add more data there, more specifically consensus integrity hash, so that we can have some alerting when it diverges (we wanted this for a while now)

Reply:
The plan is to make sure others can't submit checkpoint messages on your behalf.
Next PR has specific TODO for it: https://github.com/MystenLabs/sui/pull/6032/files#diff-ed86b8167f6985d5e9921bba3b99e5d28ba89a2168f5c071b968ee695d93d974R366

Reply:
I still don't get it: to verify that you can just compare the checkpoint summary signature with cert author. There is no need for any other data.